torch/quantization/fx/_equalize.py

```diff
@@ -224,3 +212,259 @@ def node_supports_equalization(node: Node, modules) -> bool:
+    equalization observer if it has been newly created
+    """
+
+    # Find the op node that comes directly after the input equaliation observer
+    op_node = None
+    for user in input_eq_obs_node.users.keys():
+        if node_supports_equalization(user, modules):
+            op_node = user
+            break
+
+    if op_node is None:
+        return None, None
+
+    elif op_node.op == 'call_module':
+        # If the next op_node is a nn.Linear layer, then it must have a
+        # WeightEqualizationObserver configuration
+        equalization_qconfig_map: Dict[str, Any] = model._equalization_qconfig_map  # type: ignore[assignment]
+        assert(equalization_qconfig_map.get(op_node.name, None) is not None)
+        weight_eq_obs = equalization_qconfig_map.get(op_node.name, None).weight()
+
+        assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
+        return op_node, weight_eq_obs
+
+    elif op_node.op == 'call_function':
+        # TODO
+        return None, None
+
+    return None, None
+
+def maybe_get_next_input_eq_obs(node: Node, modules: Dict[str, nn.Module]) -> Optional[_InputEqualizationObserver]:
+    """ Gets the following input equalization observer if it exists.
+
+    For example, in the case of connecting linear layers:
+        x -> inp_obs1 -> eq_obs1 -> linear1 -> out_obs1 -> eq_obs2 -> linear2 -> out_obs2
+    If the node being passed in is the linear1 node, then we want to return eq_obs2,
+    the following equalization observer for linear2.
+
+    However, if there are no connecting layers:
+        x -> inp_obs1 -> eq_obs1 -> linear1 -> out_obs1 -> add
+    Then we want to return None.
+    """
+
+    assert((node.op == 'call_module' and isinstance(modules[str(node.target)], nn.Linear)) or
+           (node.op == 'call_function' and node.target == nn.functional.linear))
+
+    # Locate the following output observer if it exists
+    maybe_obs_node = maybe_get_next_module(node, modules, ObserverBase)
+    if maybe_obs_node is None:
+        return None
+
+    maybe_eq_obs_node = maybe_get_next_module(maybe_obs_node, modules, _InputEqualizationObserver)
+    if maybe_eq_obs_node is None:
+        return None
+
+    maybe_eq_obs = modules[str(maybe_eq_obs_node)]
+    assert(isinstance(maybe_eq_obs, _InputEqualizationObserver))
+    return maybe_eq_obs
+
+def maybe_get_next_equalization_scale(node: Node, modules: Dict[str, nn.Module]) -> Optional[torch.Tensor]:
+    """ If the next next node is an InputEqualizationObserver then we want to
+    return its equalization scale, else we return 1
+
+    This is used in the case where there are two connecting linear layers:
+        linear1 -> LinearOutObs -> InputEqObs -> linear2
+    In this case, the node given is linear1 and we want to locate the InputEqObs.
+    """
+    next_inp_eq_obs = maybe_get_next_input_eq_obs(node, modules)
+    if isinstance(next_inp_eq_obs, _InputEqualizationObserver):
```

Comment:
nit: since the previous function only returns a value when it finds an _InputEqualizationObserver, maybe we can just check `if next_inp_eq_obs` here?